src/components/button/ProgressBar.js

```diff
@@ -0,0 +1,134 @@
+import React, { Component, createRef } from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import { random } from 'lodash';
+import onTransitionEnd from 'proper-on-transition-end';
+import styles from './ProgressBar.css';
+
+const INCREMENT_INTERVAL = 300;
+const MAX_PROGRESS = 0.95;
+
+class ProgressBar extends Component {
+    progressBarRef = createRef();
+
+    componentDidMount() {
+        this.maybeHandleRunningChange(false);
+    }
+
+    componentDidUpdate(prevProps) {
+        this.maybeHandleRunningChange(prevProps.running);
```

Comment:
Same as the Button component, should we change it to `handleRunningChange` ?

Reply:
👍